fs/ggml/ggml.go

```diff
@@ -141,16 +190,24 @@ func (kv KV) OllamaEngineRequired() bool {
 }
 
 func keyValue[T string | uint32 | uint64 | float32 | *array | bool](kv KV, key string, defaultValue ...T) T {
+	if val, ok := keyValueUntyped(kv, key); ok {
+		return val.(T)
+	}
+
+	slog.Warn("key not found", "key", key, "default", defaultValue[0])
+	return defaultValue[0]
+}
+
+func keyValueUntyped(kv KV, key string) (any, bool) {
```

Comment:
This is fine for now but this is a weird pattern

Reply:
the main problem seems to be array values are untyped (any) #10382 changes it to a generic and updates `keyValue` to take types `*array[string] | *array[uint32] | *array[float32] | *array[bool]`. For your use case, you can update `keyValue` to return `T, bool` where the bool indicates the type assertion result. This allows the caller to optionally try other types

e.g.

```go
if f32, ok := keyValue(kv, "key", 1.); ok {
    // key is a float32
} else if f32s, ok := keyValue(kv, "key", []float32{1.}); ok {
    // key is a float32 slice
} else {
    // default
}
```